src/search/ir.h

```diff
@@ -106,16 +116,23 @@ struct TagContainExpr : BoolAtomExpr {
 
   NodeIterator ChildBegin() override { return {field.get(), tag.get()}; };
   NodeIterator ChildEnd() override { return {}; };
+
+  std::unique_ptr<Node> Clone() const override {
+    return std::make_unique<TagContainExpr>(Node::MustAs<FieldRef>(field->Clone()),
+                                            Node::MustAs<StringLiteral>(tag->Clone()));
+  }
 };
 
-struct NumericLiteral : Node {
+struct NumericLiteral : Literal {
```

Comment:
Btw, do we have a Literal class for all literals(which is not a node), which could be used in filters?

Reply:
Currently no. Is it necessary?